Code/Tools/AssetProcessor/native/AssetManager/AssetRequestHandler.cpp

```diff
@@ -72,6 +73,60 @@ namespace
     using namespace AzToolsFramework::AssetSystem;
     using namespace AzFramework::AssetSystem;
 
+    AssetChangeReportResponse HandleAssetChangeReportRequest(MessageData < AssetChangeReportRequest> messageData)
+    {
+        AZStd::vector<AZStd::string> lines;
+
+        auto* relocationInterface = AZ::Interface<AssetProcessor::ISourceFileRelocation>::Get();
+        if (relocationInterface)
+        {
+            switch (messageData.m_message->m_type)
+            {
+                case AssetChangeReportRequest::ChangeType::CheckMove:
+                {
+                    auto resultCheck = relocationInterface->Move(messageData.m_message->m_fromPath, messageData.m_message->m_toPath);
+
+                    if (resultCheck.IsSuccess())
+                    {
+                        AssetProcessor::RelocationSuccess success = resultCheck.TakeValue();
+
+                        // The report can be too long for the AZ_Printf buffer, so split it into individual lines
+                        AZStd::string report =
+                            relocationInterface->BuildChangeReport(success.m_relocationContainer, success.m_updateTasks);
+                        AzFramework::StringFunc::Tokenize(report.c_str(), lines, "\n");
+
+                        for (const AZStd::string& line : lines)
+                        {
+                            AZ_Printf(AssetProcessor::ConsoleChannel, (line + "\n").c_str());
+                        }
+                    }
+                }
+                break;
+                case AssetChangeReportRequest::ChangeType::Move:
+                    {
+                        auto resultMove = relocationInterface->Move(messageData.m_message->m_fromPath, messageData.m_message->m_toPath, false, true, true, true);
+
+                        if (resultMove.IsSuccess())
+                        {
+                            AssetProcessor::RelocationSuccess success = resultMove.TakeValue();
+
+                            // The report can be too long for the AZ_Printf buffer, so split it into individual lines
+                            AZStd::string report =
+                                relocationInterface->BuildChangeReport(success.m_relocationContainer, success.m_updateTasks);
+                            AzFramework::StringFunc::Tokenize(report.c_str(), lines, "\n");
+
+                            for (const AZStd::string& line : lines)
+                            {
+                                AZ_Printf(AssetProcessor::ConsoleChannel, (line + "\n").c_str());
+                            }
+                        }
+                    }
+                    break;
+                }
```

Comment:
Nitpick: looks like there is an extra indent on these lines.

Reply:
Fixed.